Libraries/Connect/Interfaces/MethodDescriptor.swift

```diff
@@ -0,0 +1,26 @@
+// Copyright 2022-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Contains metadata on a specifific RPC method.
+public struct MethodDescriptor: Equatable {
+    /// The name of the RPC method (1:1 with the `.proto` file).
+    public let name: String
+    /// The path of the RPC, constructed using the package, service, and RPC name.
+    public let path: String
```

Comment:
I know nothing about how sophisticated Swift's string interning is, but to a Swift noob it seems like we may be missing an opportunity to cut some binary size here.

I assume that identical string constants get deduplicated. If so, we might be able to get a small win with this instead?

```swift
public struct MethodDescriptor: Equatable {
    public let name: String
    let service: String
    
    public var path: String {
        return self.service + "/" + self.name
    }

    public init(service: String, name: String) {
        self.name = name
        self.service = service
    }
}
```

The service will be the same for many methods (and is usually the bulk of the path), so hopefully we save some space?

Reply:
SGTM 👍🏽